core/metadata-3.2.md

```diff
@@ -20,26 +40,139 @@ This proposal aims to codify one mechanism for working with metadata: metadata
 may be configured through a client-to-server event created specifically for
 this purpose.
 
-## METADATA
+## Mechanisms
+
+This document defines the following new protocol features:
+
+* Capability: `metadata`
+* Server notification: `METADATA`
+* Client command: `METADATA`
+* Server reply and error numerics
+
+### Capability
+
+The `metadata` capability and value indicates that a server supports metadata, and communicates any configuration value keys that may affect the level of support.
+
+The ABNF format of the `metadata` capability and value is: (TODO verify ABNF)
+
+    capability ::= 'metadata' ['=' value]
+    value      ::= value_item [',' value_item]*
+    value_item ::= ( 'maxsub=' integer ) |  ( 'maxkey=' integer )
+
+The value keys are defined as follows:
+
+* `maxsub`: the maximum number of keys a client is allowed in its subscripion list. See the [Metadata subscriptions](#todo) section for more details.
+* `maxkey`: the maximum number of keys a client is allowed to set on its own nickname.
+
+
+TODO is this still required? include it for back compat?
+If `METADATA` is supported, it MUST be specified in `RPL_ISUPPORT` using the
+`METADATA` key. Servers MAY specify a limit on the number of explicitly-set
+keys per-user; the format in that case MUST be `METADATA=<integer>`, where
+`<integer>` is the limit.
+
+
+### Server notification
+
+Clients that request the `metadata` capability MUST be able to handle the `METADATA` server notification. After negotiating this capability, servers MAY send these notifications to clients at any time.
+
+The format of the `METADATA` server notication is:
+
+    METADATA <Target> <Key> <Visibility> <Value>
+
+`Target` MUST be any valid nickname or channel name.
+
+`Key` MUST be a valid key name
+
+`Visibility` MUST be an asterisk (`*`) for keys visible to everyone, or an implementation-defined value which describes the key's visibility status; for instance, it MAY be a permission level or flag. (TODO should we define a prefix like STATUSMSG for this value?)
 
-All metadata subcommands will flow through the outgoing `METADATA` verb.
+`Value` MUST be any UTF-8 encoded value.
```

Comment:
"MUST be any" sounds a bit weird. can we just simplify this down to just _"`Value` MUST be in UTF-8 encoding"_?